frame/executive/src/lib.rs

```diff
@@ -821,7 +821,9 @@ mod tests {
 	}
 
 	fn new_test_ext(balance_factor: Balance) -> sp_io::TestExternalities {
+		use sp_core::storage::well_known_keys::HEAP_PAGES;
 		let mut t = frame_system::GenesisConfig::default().build_storage::<Runtime>().unwrap();
+		t.top.insert(HEAP_PAGES.to_vec(), 8u64.encode());
```

Comment:
Where does the 8 literal come from? Is it worth extracting to a named constant?

Reply:
So this comes from _primitives/state-machine/src/testing.rs_. In `TestExternalities::new_with_code` we remove 

```
storage.top.insert(HEAP_PAGES.to_vec(), 8u64.encode());
```

Without removing that, we always used 8 heap pages, regardless of the value we gave for `heap_pages` to `NativeExecutor:::new`. (TBH I am not exactly sure why our `heap_pages` arg wouldn't override).

Since some tests that implicitly relied on heap pages being 8 broke I added it back in. However, I agree its obscure to the reader. I will try adding it as a constant to _primitives/state-machine/src/testing.rs_

Reply:
lmk if you think the changes in https://github.com/paritytech/substrate/pull/8966/commits/0e8975b5e011d9456b542b4455f46bdc831eebfc are helpful

Reply:
Yeah, that makes things clearer.